web/src/stream_data.js

```diff
@@ -860,3 +862,21 @@ export function get_options_for_dropdown_widget() {
         }))
         .sort((a, b) => util.strcmp(a.name.toLowerCase(), b.name.toLowerCase()));
 }
+
+export function user_is_topic_participant(stream_id, topic, current_user_id) {
```

Comment:
So this function is not a correct calculation for two reasons:
* Whether the mention targets the current user for other notification purposes was determined at the time the message was sent/edited, and is somewhat recorded in the message flags; but this doesn't inspect them.
* It only looks at messages that we have locally, so older participation is not accounted for.

I wonder if there's a way to avoid needing this function? Hmm, it's possible that we need to add a `topic_mentioned` `UserMessage` flag for whether the message contains a topic mention that triggered for the current user, and then we'd be able to have a simple rule that we highlight `@topic` in a message if and only if that flag is true? Would be a data model/API change but maybe a relatively cheap one, since these should be fairly rare.